src/pqm/pqm.py

```diff
@@ -216,36 +567,58 @@ def pqm_pvalue(
         For best results, we suggest using a large number of re-tessellations,
         though this is our recommendation in any case.
 
+
     Returns
     -------
     float or list
         pvalue(s). Null hypothesis that both samples are drawn from the same
         distribution.
     """
+    # Check the device and convert to the respective type (Numpy or Torch) and call their respective _pqm_test function
+
+    if device.type == 'cpu':
+        # Check if x_samples and y_samples are not already NumPy arrays
+        if not isinstance(x_samples, np.ndarray):
+            x_samples = x_samples.cpu().numpy()
+        if not isinstance(y_samples, np.ndarray):
+            y_samples = y_samples.cpu().numpy()
+    elif device.type == 'cuda':
+        # Check if x_samples and y_samples are not already torch tensors
+        if not torch.is_tensor(x_samples):
+            x_samples = torch.tensor(x_samples, device=device)
+        if not torch.is_tensor(y_samples):
+            y_samples = torch.tensor(y_samples, device=device)
+
     if re_tessellation is not None:
         return [
             pqm_pvalue(
                 x_samples,
                 y_samples,
                 num_refs=num_refs,
-                whiten=whiten,
+                z_score_norm=z_score_norm,
                 x_frac=x_frac,
                 gauss_frac=gauss_frac,
+                device=device,
             )
             for _ in range(re_tessellation)
         ]
-    _, pvalue, _, _ = _pqm_test(x_samples, y_samples, num_refs, whiten, x_frac, gauss_frac)
-    return pvalue
-
+    
+    _, p_value, _, _ = _pqm_test(
+        x_samples, y_samples, num_refs, z_score_norm, x_frac, gauss_frac, device
+    )
+    
+    # Return p-value as a float
+    return p_value if isinstance(p_value, float) else float(p_value)
```

Comment:
I think `p_value` will always be a float, so this check isn't needed

Reply:
I am inclined to keep this just in case, I don't see the reason to remove it